.github/workflows/docker-test-and-push.yaml

```
@@ -1,92 +1,101 @@
-          # Run tests that depend on data model. Tests that require demo data are ignored and run below in qgep_build_pum
-          docker exec -e PGSERVICE=qgep_build qgep pytest --ignore test/test_import.py --ignore test/test_swmm.py
-        env:
-          PGIS: ${{ matrix.pgis }}
-
-      - name: "run tests on qgep_build_pum (production)"
-        run: |
-          docker start qgep
-          docker exec -e PGSERVICE=qgep_build_pum qgep pytest
-        env:
-          PGIS: ${{ matrix.pgis }}
-
-      - name: "dockerhub login"
-        if: github.event_name != 'pull_request'
-        env:
-          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
-          DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
-        run: |
-          docker login --username "$DOCKERHUB_USERNAME" --password "$DOCKERHUB_PASSWORD"
-
-      - name: "push to dockerhub"
-        if: github.event_name != 'pull_request'
-        env:
-          PGIS: ${{ matrix.pgis }}
-        run: |
-          docker push opengisch/qgep_datamodel:${GITHUB_REF##*/}-${PGIS}
-
-      - name: "push to dockerhub (:tag)"
-        if: github.event_name != 'pull_request' && matrix.pgis == env.pgis_stable && github.ref != 'refs/heads/master'
-        env:
-          PGIS: ${{ matrix.pgis }}
-        run: |
-          docker tag opengisch/qgep_datamodel:${GITHUB_REF##*/}-${PGIS} opengisch/qgep_datamodel:${GITHUB_REF##*/}
-          docker push opengisch/qgep_datamodel:${GITHUB_REF##*/}
-
-      - name: "push to dockerhub (:latest)"
-        if: github.event_name != 'pull_request' && github.ref == 'refs/heads/master'
-        env:
-          PGIS: ${{ matrix.pgis }}
-        run: |
-          docker tag opengisch/qgep_datamodel:${GITHUB_REF##*/}-${PGIS} opengisch/qgep_datamodel:latest
-          docker push opengisch/qgep_datamodel:latest
-
-      - name: "failure logs"
-        if: failure()
-        run: |
-          docker logs qgep
+name: Docker
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  docker-tests:
+    name: Docker
+    runs-on: ubuntu-18.04
+
+    strategy:
+      matrix:
+        # postgres-postgis version, see available tags https://hub.docker.com/r/postgis/postgis/tags
+        pgis: ["9.6-2.5", "10-2.5", "11-2.5", "12-2.5", "9.6-3.0", "10-3.0", "11-3.0", "12-3.0"]
+      fail-fast: false
+
+    env:
+      # which pgis version to use for :tag images (must be in the matrix above)
+      pgis_stable: "11-2.5"
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: "assert version is up to date"
+        run: |
+          python -c '
+          import os
+          cur_version = open("system/CURRENT_VERSION.txt","r").read().strip()
+          last_delta_version = sorted(list([d for d in os.listdir("delta") if d.startswith("delta_")]))[-1].split("_")[1]
+          assert cur_version == last_delta_version, cur_version + " != " + last_delta_version
+          '
```

```suggestion
          LAST_VERSION=$(ls delta | sort -r | sed '/^delta_/!d' | head -n 1 | cut -d_ -f2)
          CUR_VERSION=$(cat system/CURRENT_VERSION.txt)
          [[ "${LAST_VERSION}" != "${CUR_VERSION}" ]] && echo "Version mismatch ${LAST_VERSION} != ${CUR_VERSION}" && exit 1
```

I tend to prefer native scripting ;)

also, are the white spaces removal intentional in the rest of the file?

---

Ok made the change and reverted unintentional whitespace